src/EditorFeatures/CSharpTest/BlockCommentCompletion/BlockCommentCompletionTests.cs

```diff
@@ -0,0 +1,346 @@
+";
+            await VerifyAsync(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.BlockCommentCompletion)]
+        public async Task InsertOnMiddleLine1()
+        {
+            var code = @"
+    /*
+     *$$*/
+";
+            var expected = @"
+    /*
+     *
+     $$*/
+";
+            await VerifyAsync(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.BlockCommentCompletion)]
+        public async Task InsertOnMiddleLine2()
+        {
+            var code = @"
+    /*
+     *$$ */
+";
+            var expected = @"
+    /*
+     *
+     *$$ */
+";
+            await VerifyAsync(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.BlockCommentCompletion)]
+        public async Task InsertOnMiddleLine3()
+        {
+            var code = @"
+    /*
+     * $$ 1.
+     */
+";
+            var expected = @"
+    /*
+     * 
+     * $$ 1.
+     */
+";
+            await VerifyAsync(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.BlockCommentCompletion)]
+        public async Task InsertOnMiddleLine4()
+        {
+            var code = @"
+    /*
+     *  1.$$
+     */
+";
+            var expected = @"
+    /*
+     *  1.
+     *  $$
+     */
+";
+            await VerifyAsync(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.BlockCommentCompletion)]
+        public async Task InsertOnMiddleLine5()
+        {
+            var code = @"
+    /*
+     *   1.
+     *   $$
+     */
+";
+            var expected = @"
+    /*
+     *   1.
+     *   
+     *   $$
+     */
+";
+            await VerifyAsync(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.BlockCommentCompletion)]
+        public async Task InsertOnMiddleLine6()
+        {
+            var code = @"
+    /*
+  $$   *
+     */
+";
+            var expected = @"
+    /*
+  
+     * $$   *
+     */
+";
+            await VerifyAsync(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.BlockCommentCompletion)]
+        public async Task InsertOnMiddleLine7()
+        {
+            var code = @"
+    /*
+     *************$$
+     */
+";
+            var expected = @"
+    /*
+     *************
+     * $$
+     */
+";
+            await VerifyAsync(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.BlockCommentCompletion)]
+        public async Task InsertOnEndLine0()
+        {
+            var code = @"
+    /*
+     *$$/
+";
+            var expected = @"
+    /*
+     *
+     *$$/
+";
+            await VerifyAsync(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.BlockCommentCompletion)]
+        public async Task InsertOnEndLine1()
+        {
+            var code = @"
+    /*
+  $$   */
+";
+            var expected = @"
+    /*
+  
+     * $$   */
```

Comment:
Corner and controversial case. VS2013 has similar behavior, but it also trim the spaces on the left of `*/`.

Isn't that the previous line starts with `/*`?